### PR TITLE
API: referring to duplicate level name raises ValueError

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
follow-up on: https://github.com/pandas-dev/pandas/pull/21423

But, this currently has a lot of failures, because of https://github.com/pandas-dev/pandas/issues/21677 that we first need to resolve.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Changed the implementation a bit to keep the fallback logic for integer names, to decouple it from the discussion in #21677

**Review by jreback — CHANGES_REQUESTED** [review]
wow much simpler 

minor comment - pls add whatsnew note

any downsides to doing this?

**Comment by jorisvandenbossche** [comment]
> wow much simpler

I don't think the code is simpler, the diff is just smaller :-)

> any downsides to doing this?

I don't think so. 
It's an api change, but only a minor one I think. In principle we could also keep it as KeyError (I don't think there is necessarily a more correct one, although KeyError is typically used for non-existing key, while here the key is existing but duplicate), but then encountering this in eg `stack` would change from ValueError to KeyError (so a api change as well).

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -229,6 +229,9 @@ Other API Changes
 ^^^^^^^^^^^^^^^^^
 
 - :class:`DatetimeIndex` now accepts :class:`Int64Index` arguments as epoch timestamps (:issue:`20997`)
+- Accessing a level of a ``MultiIndex`` with a duplicate name (e.g. in
+  :meth:~MultiIndex.get_level_values) now raises a ValueError instead of
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: double backticks around the errors

**Review by jreback — CHANGES_REQUESTED** [review]
minor comment

**Review by jreback — APPROVED** [review]

**Pull request merged**